tests/integration/devfile/cmd_dev_test.go

```diff
@@ -0,0 +1,160 @@
+			helper.WaitForOutputToContain(watch.PushErrorString, 180, 10, session)
+		})
+		It("should use the index information from previous push operation", func() {
+			// Create a new file A
+			fileAPath, fileAText := helper.CreateSimpleFile(commonVar.Context, "my-file-", ".txt")
+
+			// watch that project
+			session := helper.CmdRunner("odo", "dev")
+			defer session.Kill()
+
+			helper.WaitForOutputToContain("Waiting for something to change", 180, 10, session)
+
+			// Change some other file B
+			helper.ReplaceString(filepath.Join(commonVar.Context, "server.js"), "App started", "App is super started")
+			helper.WaitForOutputToContain("Executing devrun command", 180, 10, session)
+
+			podName := commonVar.CliRunner.GetRunningPodNameByComponent(cmpName, commonVar.Project)
+
+			// File should exist, and its content should match what we initially set it to
+			execResult := commonVar.CliRunner.Exec(podName, commonVar.Project, "cat", "/projects/"+filepath.Base(fileAPath))
+			Expect(execResult).To(ContainSubstring(fileAText))
+		})
+		It("ensure that index information is updated", func() {
+			indexAfterPush, err := util.ReadFileIndex(filepath.Join(commonVar.Context, ".odo", "odo-file-index.json"))
+			Expect(err).ToNot(HaveOccurred())
+
+			// watch that project
+			session := helper.CmdRunner("odo", "dev")
+			defer session.Kill()
+
+			helper.WaitForOutputToContain("Waiting for something to change", 180, 10, session)
+
+			// Create a new file A
+			fileAPath, _ := helper.CreateSimpleFile(commonVar.Context, "my-file-", ".txt")
+
+			// Wait for the new file to exist in the index
+			Eventually(func() bool {
+
+				newIndexAfterPush, readErr := util.ReadFileIndex(filepath.Join(commonVar.Context, ".odo", "odo-file-index.json"))
+				if readErr != nil {
+					fmt.Fprintln(GinkgoWriter, "New index not found or could not be read", readErr)
+					return false
+				}
+
+				_, exists := newIndexAfterPush.Files[filepath.Base(fileAPath)]
+				if !exists {
+					fmt.Fprintln(GinkgoWriter, "path", fileAPath, "not found.", readErr)
+				}
+				return exists
+
+			}, 180, 10).Should(Equal(true))
+
+			// Delete file A and verify that it disappears from the index
+			err = os.Remove(fileAPath)
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(func() bool {
+
+				newIndexAfterPush, err := util.ReadFileIndex(filepath.Join(commonVar.Context, ".odo", "odo-file-index.json"))
+				if err != nil {
+					fmt.Fprintln(GinkgoWriter, "New index not found or could not be read", err)
+					return false
+				}
+
+				// Sanity test: at least one file should be present
+				if len(newIndexAfterPush.Files) == 0 {
+					return false
+				}
+
+				// The fileA file should NOT be found
+				match := false
+				for relativeFilePath := range newIndexAfterPush.Files {
+
+					if strings.Contains(relativeFilePath, filepath.Base(fileAPath)) {
+						match = true
+					}
+				}
```

Comment:
`newIndexAfterPush` is a map of file names and file data, so I'm sure you must have thought of this, but I am wondering if we can hardcode the filename and simply check if the key exists.

Reply:
I haven't thought of it because I copied the tests from `odo watch` command. But `newIndexAfterPush` is the index created by calling `ReadFileIndex` function. I feel it's not a good idea to hard code it because we can't predict what exactly would be there in the index file. Not without spending a lot of time, and that won't be worth it, I think.

Let me know if you think otherwise because, as I said, I copied over the tests from `odo watch`.